### ISSUE TITLE
[FEA] Support additional merge keywords left/right_on, indicator, suffixes

### ISSUE THREAD

**Issue by mrocklin** [rapidsai/cudf#1031]
If cudf can support the following keywords to merge then we can start to reuse some of the corner case implementations in dask.dataframe.merge.  In particular I'm looking at the large dataframe to single-partition dataframe case.

-  `suffixes`: This seems to have replaced `lsuffix` and `rsuffix` currently in cudf.DataFrame.merge.  Those keywords no longer seem to be present in the Pandas signature
-  `right_on` and `left_on`: these allow different columns on the left and right side to be used.  My guess is that this would be generally useful beyond just Dask support
-  `indicator`: honestly I don't ever use this (nor do I care that it is implemented, but it would be useful for it to be in the signature at least, even if it just raises a `NotImplementedError` if the value is not None.

cc @thomcom @randerzander

**Comment by thomcom** [comment]
It looks like you need `indicator` and `suffixes` for 0.6, correct? Those are the last currently known items for this feature tree?

**Comment by mrocklin** [comment]
I think so, and I don't really need them, I just need them to be keyword arguments that won't err in the common case that ..

- indicator is the default value

    ```python
    def merge(..., indicator=False, ...):
        if indicator:
            raise NotImplementedError(...)
   ```

-   suffixes just expands to `lsuffix`/`rsuffix`
    
    ```python
    def merge(..., suffixes=(..., ...), ...):
        lsuffix, rsuffix = suffixes

    ```

    (though this might have backwards compatibility concerns with cudf, not sure)

**Comment by kkraus14** [comment]
> * (though this might have backwards compatibility concerns with cudf, not sure)

Given the immaturity of our project, we aren't guaranteeing backwards compatibility between releases yet, so we should strive for current Pandas behavior.

**Comment by mrocklin** [comment]
@randerzander if you apply the following diff then the indexed_joins in https://github.com/rapidsai/dask-cudf/pull/91 work

```diff
diff --git a/python/cudf/dataframe/dataframe.py b/python/cudf/dataframe/dataframe.py
index a0bb5a0..923a313 100644
--- a/python/cudf/dataframe/dataframe.py
+++ b/python/cudf/dataframe/dataframe.py
@@ -1334,7 +1334,7 @@ class DataFrame(object):

     def merge(self, right, on=None, how='inner', left_on=None, right_on=None,
               left_index=False, right_index=False, lsuffix='_x', rsuffix='_y',
-              type="", method='hash'):
+              type="", method='hash', suffixes=('_x', '_y'), indicator=False):
         """Merge GPU DataFrame objects by performing a database-style join
         operation by columns or indexes.
```

If there are other configurations that you care about I encourage you to raise them on that PR

**Comment by kkraus14** [comment]
Fixed by #1091
